src/algebra/add_comm_torsor.lean

```diff
@@ -0,0 +1,180 @@
+/-
+Copyright (c) 2020 Joseph Myers. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Joseph Myers.
+-/
+import algebra.group.basic
+
+/-!
+# Torsors of additive commutative group actions
+
+This file defines torsors of additive commutative group actions.
+
+## Notations
+
+The group elements are referred to as vectors and the elements they
+act on are referred to as points, reflecting the use case of affine
+spaces.  This file defines the notation `+ᵥ` for adding a vector to a
+point and `-ᵥ` for subtracting two points to produce a vector.
+
+## Implementation notes
+
+It may be appropriate to refactor in future as a special case of
+torsors of additive group actions, in terms of the general definition
+of group actions (currently mathlib only has multiplicative group
+actions).  The variable `V` is an explicit rather than implicit
+argument to lemmas because otherwise the elaborator has problems
+inferring appropriate types and type class instances.
+
+## References
+
+* https://en.wikipedia.org/wiki/Principal_homogeneous_space
+* https://en.wikipedia.org/wiki/Affine_space
+
+-/
+
+/-- Type class for the `+ᵥ` and `-ᵥ` notation. -/
+class has_vadd (V : Type*) (P : Type*) :=
+(vadd : P → V → P)
+(vsub : P → P → V)
+
+infix `+ᵥ`:65 := has_vadd.vadd
+infix `-ᵥ`:65 := has_vadd.vsub
+
+section prio
+set_option default_priority 100 -- see Note [default priority]
```

Comment:
Could you add the line
```lean
set_option old_structure_cmd true
```
here? (To make sure that you get a flat structure, and not a substructure containing the `has_vadd` field).

Reply:
I've made that change.